kalite/control_panel/views.py

```diff
@@ -141,12 +141,14 @@ def zone_management(request, zone_id="None"):
             "num_users":  FacilityUser.objects.filter(facility=facility).count(),
             "num_groups": FacilityGroup.objects.filter(facility=facility).count(),
             "id": facility.id,
+            "meta_data_in_need": check_meta_data(facility.id),
```

Comment:
Pass in the facility here, not the facility.id, as we have the model already referenced and available. Otherwise you are forcing a database lookup on every metadata check (so this will scale linearly with the number of facilities).